src/bootstrap/bin/rustc.rs

```diff
@@ -93,6 +93,21 @@ fn main() {
              env::join_paths(&dylib_path).unwrap());
     let mut maybe_crate = None;
 
+    // Get the name of the crate we're compiling, if any.
+    let maybe_crate_name = args.windows(2)
+        .find(|a| &*a[0] == "--crate-name")
```

Comment:
please add an assert for checking whether the `args` contain `--crate-name=something`, since that will get silently ignored here.

Reply:
I'd rather not add more noise to this file.